synapse/federation/sender/__init__.py

```diff
@@ -434,6 +434,17 @@ async def handle_event(event: EventBase) -> None:
                         # If there are no prev event IDs then the state is empty
                         # and so no remote servers in the room
                         destinations = set()
+                    elif await self.store.is_partial_state_room(event.room_id):
```

Comment:
perhaps we should not do this check and instead directly call `get_partial_state_servers_at_join`, and consider that we are not partially join if the set is empty, as per doc of the method.

This would avoid a round trip to the DB but we would do a single query potentially a bit more heavy. Not sure what is better.

Reply:
There's a race here if we do it in two db queries. If the room gets un-partial stated after the `is_partial_state_room`, but before `get_partial_state_servers_at_join`, we will read off an empty destination list.